igvm/vm.py

```diff
@@ -713,3 +714,75 @@ def unblock_autostart(self):
 
     def copy_postboot_script(self, script):
         self.put('/buildvm-postboot', script, '0755')
+
+    def aws_disk_set(self, size: int) -> None:
+        """AWS disk set
+
+        Resize a disk in AWS.
+
+        :param: size: New disk_size
+
+        :raises: VMError: Generic exception for VM errors of all kinds
+        """
+
+        ec2 = boto3.resource('ec2')
+
+        response = ec2.Instance(self.dataset_obj['aws_instance_id'])
+        for vol in response.volumes.all():
+            volume_id = vol.id
+            break
+
+        ec2 = boto3.client('ec2')
+        ec2.modify_volume(VolumeId=volume_id, Size=int(size))
+
+        time.sleep(5)
```

Comment:
This is risky, wouldn't it be better to check disk size in a loop and wait for it to really grow?

Reply:
Changed to while loop, which is checking the actual disk_size with `lsblk -bdnro size` before trying to increase the filesystem.